src/packages/cli/src/bin.ts

```diff
@@ -198,6 +209,51 @@ async function main(): Promise<number> {
   return 0
 }
 
+/**
+ * Get a unique identifier for the project by hashing
+ * the directory with `schema.prisma`
+ */
+async function getProjectHash(): Promise<string> {
+  const schemaPath = await getSchemaPath()
+
+  return crypto
+    .createHash('sha256')
+    .update(schemaPath)
+    .digest('hex')
+    .substring(0, 8)
+}
+/**
+ * Get a unique identifier for the CLI instllation path
+ * which can be either global or local (in project's node_modules)
+ */
+async function getCLIPathHash(): Promise<string> {
+  const cliPath = process.argv[1]
+  return crypto
+    .createHash('sha256')
+    .update(cliPath)
+    .digest('hex')
+    .substring(0, 8)
+}
+
+/**
+ * Get the path where `schema.prisma` lives
+ */
+async function getSchemaPath(): Promise<string> {
```

Comment:
It returns a string not a Promise?

Reply:
Since it's async for the await calls, it does return a Promise.